tools/code_analyzer/gen_op_registration_whitelist.py

```diff
@@ -42,6 +42,10 @@ def gen_transitive_closure(dep_graph, root_ops):
     result = set(root_ops)
     queue = root_ops[:]
 
+    # The dependency graph might contain a special entry with key = `__BASE__`
+    # and value = (set of `base` ops to always include in custom build).
+    queue.append('__BASE__')
```

Comment:
Should we insist that the set of base ops is closed? IIUC adding them here lets us handle a base op generating a non-base dependency, but do we need/want to do that?

Reply:
@bhosmer maybe, having an explicit closure can prevent us from inadvertently include more ops than what we expect; but on the other hand I feel we will need take extra step to verify the base ops is a closure (to avoid more subtle runtime error). I lean towards the current approach as including too many ops can be captured by build size bot.

Reply:
Yeah, it's true that verifying would be an explicit step - basically a preliminary run through the loop that follows. (And definitely agreed that letting a missed dependency cause a runtime error is suboptimal :P) Sounds like the build size bot is the safety net, so maintaining a closure invariant is probably overkill.